insights/parsers/named_conf.py

```diff
@@ -16,31 +16,44 @@
 class NamedConf(NamedCheckconf):
     """
     Class for parsing the file ``/etc/named.conf```, We use class ``NamedCheckConf`` to parse most
-    of the named.conf configurations and class ``NamedConf`` to parse the `include` directives.
+    of the named.conf configurations and class ``NamedConf`` to parse the `include` directives and
+    the `allow-recursion` value from the `options` section.
 
     .. note::
         Please refer to the super-class :py:class:`insights.parsers.named_checkconf:NamedCheckConf`
         for more usage information.
 
     Attributes:
-        includes (list): List of files in 'include' section.
+        includes (list): List of files in 'include' and 'allow-recursion' value from the 'options' section.
 
     Raises:
         SkipException: When content is empty or cannot be parsed.
 
     Examples:
         >>> named_conf.includes
         ['/etc/crypto-policies/back-ends/bind.config']
+        >>> named_conf.allow_recursion_address
+        ['localnets', 'localhost', '192.168.10.1/24']
     """
 
     def parse_content(self, content):
         includes = []
+        allow_recursion_address = []
+        options_flag = False
         super(NamedConf, self).parse_content(content)
 
         try:
-            for line in [l for l in content if l.strip().startswith('include ') and ';' in l]:
-                includes.append(line.split(';')[0].replace('"', '').split()[1])
+            for line in content:
+                if line.strip().startswith('include ') and ';' in line:
+                    includes.append(line.split(';')[0].replace('"', '').split()[1])
+                if line.strip().startswith('options ') and ';' not in line:
```

Comment:
I doubt if the check of the `options` section here is required or correct.

Is the `allow-recursion` a unique directive only for the `option` section?

1.  If not, that's it may appear in other sections.
     The current code is incorrect. It's required to check the end of the `options` section, to avoid getting the `allow-recursion` from the other sections which could override the expected one.
2. If yes, it seems that it's not necessary to do this check.